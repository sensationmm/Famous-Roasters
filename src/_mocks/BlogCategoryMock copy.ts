import { GraphQLError } from 'graphql'
import { loader } from 'graphql.macro'

const GET_BLOG_CATEGORY_LIST = loader('src/graphql/queries/blogCategoryListDetails.query.graphql')

export const BlogCategoryListMock = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    data: {
      categoryList: [
        {
          __typename: 'Category',
          name: 'Kaffeewissen',
          summary: 'Werde zum Kaffee-Nerd. Unsere Experten erklären alles: Von Aroma bis Zubereitungsmethode.',
          tags: ['Nachhaltigkeit', 'SpecialityCoffee'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeeequipment',
          summary:
            'Unsere Experten testen und bewerten das beste Kaffeeequipment. Wir wollen sicherstellen, dass du die beste Hardware für einen exzellenten Kaffee hast.',
          tags: ['Handfiltervergleich', 'Grundausstattung'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeetrends',
          summary: 'Die aktuellsten Trends über die Kaffeewelt.',
          tags: ['Gadgets', 'Roestungen'],
        },
        {
          __typename: 'Category',
          name: 'Zubereitungstipps',
          summary:
            'Mach mehr aus deinem Kaffee: our experts prepared a few brewing guides to help you enjoy always the best coffee.',
          tags: ['Aeropress', 'Siebtraeger', 'ColdDrip'],
        },
      ],
    },
  },
}

export const BlogCategoryListMockError = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    errors: [new GraphQLError('Error!')],
  },
}
