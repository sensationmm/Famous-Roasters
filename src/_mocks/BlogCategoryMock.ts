import { GraphQLError } from 'graphql'
import { loader } from 'graphql.macro'

const GET_BLOG_CATEGORY_LIST = loader('src/graphql/queries/blogCategoryListDetails.query.graphql')

export const BlogCategoryListMock = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    data: {
      categoryList: [
        {
          __typename: 'Category',
          name: 'Kaffeewissen',
          summary:
            'Lever up your coffee-culture. Our expert will touch different topics: beans types, origins, cupping and many more...',
          tags: ['Nachhaltigkeit', 'SpecialityCoffee'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeeequipment',
          summary:
            'Here our coffee experts test and review different equipment. They aim to cover from basic to more niche equipment.',
          tags: ['Handfiltervergleich', 'Grundausstattung'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeetrends',
          summary:
            'Here our coffee experts test and review different equipment. They aim to cover from basic to more niche equipment.',
          tags: ['Gadgets', 'Roestungen'],
        },
        {
          __typename: 'Category',
          name: 'Zubereitungstipps',
          summary:
            'Mach mehr aus deinem Kaffee: our experts prepared a few brewing guides to help you enjoy always the best coffee.',
          tags: ['Aeropress', 'Siebtraeger', 'ColdDrip'],
        },
      ],
    },
  },
}

export const BlogCategoryListMockError = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    errors: [new GraphQLError('Error!')],
  },
}
