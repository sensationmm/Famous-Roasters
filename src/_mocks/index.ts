import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
