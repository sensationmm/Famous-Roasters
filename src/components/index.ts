import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
