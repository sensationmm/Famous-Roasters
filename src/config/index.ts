import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
