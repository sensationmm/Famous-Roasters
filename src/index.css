@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}