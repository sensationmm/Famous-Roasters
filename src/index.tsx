import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
