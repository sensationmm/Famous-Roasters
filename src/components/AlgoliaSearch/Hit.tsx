import { useTranslation } from 'react-i18next'
import { Link } from 'react-router-dom'
import { formatPrice } from 'src/utils'

import { Tag, TagType } from '../Tag'
import { Typography, TypographySize, TypographyType } from '../Typography'

type RawHit = {
  id: string
  title: string
  image: string
  product_image: string
  vendor: string
  inventory_quantity: number
  variants_inventory_count: number
  meta: {
    my_fields?: {
      decaf: boolean | string
      origin: string[]
      coffee_type: string
      price_per_kg: number
    }
  }
  variants_min_price: number
  variants_max_price: number
  variants_count: number
  inventory_management: string | undefined
}

const Hit = ({ hit }: { hit: RawHit }) => {
  const { t } = useTranslation()
  const showFrom = hit.variants_count > 1
  const {
    id,
    image,
    title,
    vendor,
    variants_inventory_count: totalInventory,
    inventory_management,
    variants_min_price,
  } = hit

  const origin = hit.meta.my_fields?.origin
  const decaf = hit.meta.my_fields?.decaf
  const coffee_type = hit.meta.my_fields?.coffee_type
  const price_per_kg = hit.meta.my_fields?.price_per_kg

  const isDecaf = decaf === true || decaf === 'true'
  const outOfStock =
    inventory_management === 'shopify' && totalInventory !== undefined && totalInventory !== null && totalInventory <= 0
  const textLineClassNames = outOfStock ? 'text-coreUI-text-tertiary' : 'text-coreUI-text-secondary'

  return (
    <Link to={`/product/${id}`} key={`product-tile-link-${id}`}>
      <div className="flex pt-8 md:px-6">
        <div className="flex justify-center items-center shrink-0 relative w-36 h-36">
          <div className="flex justify-center items-center rounded-full bg-coreUI-background-images w-36 h-36">
            <img
              src={`${image}&width=150&height=150`}
              alt={title}
              className="w-24 max-h-24"
              width="150px"
              height="150px"
            />
            {isDecaf && <Tag type={TagType.Decaf} value="Decaf" small={true} className="absolute top-2 left-0" />}
          </div>
        </div>

        <div className="flex flex-col w-full justify-start pl-4">
          <Typography
            as="div"
            type={TypographyType.Label}
            size={TypographySize.Base}
            className={outOfStock ? 'text-coreUI-text-tertiary' : ''}
          >
            {title}
          </Typography>
          <Typography
            as="div"
            type={TypographyType.Paragraph}
            size={TypographySize.Base}
            className={`flex flex-col w-full justify-start${outOfStock ? ' text-coreUI-text-tertiary' : ''}`}
          >
            {vendor}
          </Typography>
          {origin && (
            <Typography
              as="div"
              type={TypographyType.Paragraph}
              size={TypographySize.Base}
              className={textLineClassNames}
            >
              {origin.map((x, idx) =>
                idx === origin.length - 1
                  ? t(`pages.catalogue.filters.origin.values.${x}`)
                  : t(`pages.catalogue.filters.origin.values.${x}`) + ', ',
              )}
            </Typography>
          )}

          {coffee_type && (
            <Typography
              as="div"
              type={TypographyType.Paragraph}
              size={TypographySize.Base}
              className={textLineClassNames}
            >
              {coffee_type}
            </Typography>
          )}

          <div className="flex items-baseline">
            <Typography
              type={TypographyType.Label}
              size={TypographySize.Base}
              className={`mr-1${outOfStock ? ' text-coreUI-text-tertiary' : ''}`}
            >
              {showFrom && t('pages.catalogue.tile.from') + ' '}
              {formatPrice(variants_min_price)}
            </Typography>
            {price_per_kg && (
              <Typography
                type={TypographyType.Paragraph}
                size={TypographySize.Tiny}
                className={outOfStock ? 'text-coreUI-text-tertiary' : 'text-coreUI-text-secondary'}
              >
                ({formatPrice(price_per_kg)}/kg)
              </Typography>
            )}
          </div>
          {outOfStock && (
            <Typography type={TypographyType.Paragraph} size={TypographySize.Tiny} className="text-negative">
              {t('pages.product.transactional.outOfStock')}
            </Typography>
          )}
        </div>
      </div>
    </Link>
  )
}

export default Hit
