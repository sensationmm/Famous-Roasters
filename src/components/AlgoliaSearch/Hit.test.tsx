import { render, waitFor } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter as Router } from 'react-router-dom'
import { ProductMockData } from 'src/_mocks'
import { i18n } from 'src/config'

import Hit from './Hit'

describe('Hit component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <Router>
        <I18nextProvider i18n={i18n}>
          <Hit
            hit={{
              ...ProductMockData,
              image: 'https://example.com/image',
              product_image: 'https://example.com/product-image',
              inventory_quantity: 25,
              variants_inventory_count: 10,
              vendor: '19grams',
              variants_min_price: 5,
              variants_max_price: 10,
              variants_count: 10,
              inventory_management: 'shopify',
              meta: {
                my_fields: {
                  decaf: false,
                  origin: ['BR', 'CO'],
                  coffee_type: 'Filter',
                  price_per_kg: 10,
                },
              },
            }}
          />
        </I18nextProvider>
      </Router>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for decaf', async () => {
    const { container } = render(
      <Router>
        <I18nextProvider i18n={i18n}>
          <Hit
            hit={{
              ...ProductMockData,
              image: 'https://example.com/image',
              product_image: 'https://example.com/product-image',
              inventory_quantity: 25,
              variants_inventory_count: 10,
              vendor: '19grams',
              variants_min_price: 5,
              variants_max_price: 10,
              variants_count: 10,
              inventory_management: 'shopify',
              meta: {
                my_fields: {
                  decaf: true,
                  origin: ['BR', 'CO'],
                  coffee_type: 'Filter',
                  price_per_kg: 10,
                },
              },
            }}
          />
        </I18nextProvider>
      </Router>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for out of stock', async () => {
    const { container } = render(
      <Router>
        <I18nextProvider i18n={i18n}>
          <Hit
            hit={{
              ...ProductMockData,
              image: 'https://example.com/image',
              product_image: 'https://example.com/product-image',
              inventory_quantity: 0,
              variants_inventory_count: 0,
              vendor: '19grams',
              variants_min_price: 5,
              variants_max_price: 10,
              variants_count: 10,
              inventory_management: 'shopify',
              meta: {
                my_fields: {
                  decaf: false,
                  origin: ['BR', 'CO'],
                  coffee_type: 'Filter',
                  price_per_kg: 10,
                },
              },
            }}
          />
        </I18nextProvider>
      </Router>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
