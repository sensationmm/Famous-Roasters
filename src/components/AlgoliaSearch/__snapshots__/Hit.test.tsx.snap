// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hit component Renders correctly 1`] = `
<div>
  <a
    href="/product/gid://shopify/Product/6837501067347"
  >
    <div
      class="flex pt-8 md:px-6"
    >
      <div
        class="flex justify-center items-center shrink-0 self-center relative w-32 h-32"
      >
        <div
          class="flex justify-center items-center rounded-full bg-coreUI-background-images w-32 h-32"
        >
          <img
            alt="Espresso Raritäten Set"
            class="w-28 max-h-28"
            src="https://example.com/image"
          />
        </div>
      </div>
      <div
        class="flex flex-col w-full justify-start pl-4"
      >
        <div
          class="text-base leading-6 font-semibold"
        >
          Espresso Raritäten Set
        </div>
        <div
          class="flex flex-col w-full justify-start text-base leading-6"
        >
          19grams
        </div>
        <div
          class="text-coreUI-text-secondary text-base leading-6"
        >
          Brasilien, 
          Kolumbien
        </div>
        <div
          class="text-coreUI-text-secondary text-base leading-6"
        >
          Filter
        </div>
        <div
          class="flex items-baseline"
        >
          <span
            class="mr-1 text-base leading-6 font-semibold"
          >
            ab 
            5,00 €
          </span>
          <span
            class="text-coreUI-text-secondary text-xs leading-4"
          >
            (
            10,00 €
            /kg)
          </span>
        </div>
      </div>
    </div>
  </a>
</div>
`;
